File: app/src/main/java/com/example/getfood/ui/loginregister/registerfragment/RegisterMvpPresenter.java
package com.example.getfood.ui.loginregister.registerfragment;

import androidx.annotation.NonNull;

import com.example.getfood.ui.base.BaseMvpPresenter;

public interface RegisterMvpPresenter<V extends RegisterMvpView> extends BaseMvpPresenter<V> {
    void performRegistration(@NonNull String userEmail, @NonNull String password, String confirmPassword);

    void verifyUserEmail();

    void signOutUser();

    boolean isTermsAndConditionChecked(boolean checked);
}

File: app/src/main/java/com/example/getfood/ui/loginregister/registerfragment/RegisterMvpPresenter.kt
package com.example.getfood.ui.loginregister.registerfragment

import com.example.getfood.ui.base.BaseMvpPresenter

open interface RegisterMvpPresenter<V : RegisterMvpView?> : BaseMvpPresenter<V> {
    fun performRegistration(userEmail: String, password: String, confirmPassword: String)
    fun verifyUserEmail()
    fun signOutUser()
    fun isTermsAndConditionChecked(checked: Boolean): Boolean
}